easybuild/easyblocks/generic/intelbase.py

```diff
@@ -102,10 +105,41 @@ def extra_options(extra_vars=None):
             # disables TMP_PATH env and command line option
             'usetmppath': [False, "Use temporary path for installation", CUSTOM],
             'm32': [False, "Enable 32-bit toolchain", CUSTOM],
+            'components': [[], "List of components to install", CUSTOM],
         })
 
         return extra_vars
 
+    def parse_components_list(self):
+        """parse the regex in the components extra_options and select the matching components
+        from the mediaconfig.xml file in the install dir"""
+
+        mediaconfigpath = os.path.join(self.cfg['start_dir'], 'pset', 'mediaconfig.xml')
+        if not os.path.isfile(mediaconfigpath):
+            self.log.warning("Could not find pset/mediaconfig.xml in the build directory. Skipping list of components to install")
+            return
+
+        mediaconfig = read_file(mediaconfigpath)
+        available_components = re.findall("<Abbr>(?P<component>[^<]+)</Abbr>", mediaconfig, re.M)
+        self.log.debug("Intel components found: %s" % available_components)
+
+        to_install = []
+        self.log.debug("Using regex list: %s" % self.cfg['components'])
+
+        if 'ALL' in self.cfg['components']:
+            to_install = ['ALL']
+        elif 'DEFAULTS' in self.cfg['components']:
+            to_install = ['DEFAULTS']
+        else:
```

Comment:
ok, sure, but it does mean there's no way around if mediaconfig.xml is not available for some reason...

unless you allow that mediaconfig.xml is not available above, and only complain here if `available_components` is `None`